### PR TITLE
[google-cloud-cpp] use features to stop package growth

### PR DESCRIPTION
**Pull request by coryan**
We have plans to grow `google-cloud-cpp`, but we think the build times
are large enough already. This creates "features" to at least stop the
package growth as new features are added. The "default-features"
reflects the current features in the package.

- #### What does your PR fix?  

N/A

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  

No chage.

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  

Yes.

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  

Yes.

### CODE REVIEW

File: ports/google-cloud-cpp/portfile.cmake [hunk 1]
```diff
@@ -12,11 +12,16 @@ vcpkg_from_github(
 
 vcpkg_add_to_path(PREPEND "${CURRENT_HOST_INSTALLED_DIR}/tools/grpc")
 
+set(GOOGLE_CLOUD_CPP_ENABLE "${FEATURES}")
+list(REMOVE_ITEM GOOGLE_CLOUD_CPP_ENABLE "core")
+list(REMOVE_ITEM GOOGLE_CLOUD_CPP_ENABLE "googleapis")
```
Comment (JackBoosY) [inline ports/google-cloud-cpp/portfile.cmake:17]: Please note that the feature name in vcpkg.json is not necessarily the same as the feature name in source.

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: ports/google-cloud-cpp/portfile.cmake (hunk 1 above)
Comment (coryan) [reply ports/google-cloud-cpp/portfile.cmake:17]: I am one of the authors of `google-cloud-cpp` ([ref](https://github.com/googleapis/google-cloud-cpp/graphs/contributors)) and I am the primary maintainer of the `google-cloud-cpp` port.

Yes, I understand that the features in `vcpkg` and the project are not *necessarily* the same.  It is our (me and my team) *intention* to keep these in sync.  That will make both our life (as maintainers of both) easier, and will make life for users also easier (we hope).

**Review by JackBoosY — APPROVED** [review]

**Comment by JackBoosY** [comment]
Tested all feature on `x86-windows` and `x64-windows-static` successfully.

File: ports/google-cloud-cpp/vcpkg.json [hunk 2]
```diff
@@ -20,5 +21,115 @@
...
+  "features": {
+    "bigquery": {
+      "description": "The Google Cloud BigQuery C++ client library",
+      "dependencies": [
+        "abseil",
+        "grpc",
+        {
+          "name": "grpc",
+          "host": true
+        },
```
Comment (ras0219-msft) [inline ports/google-cloud-cpp/vcpkg.json:42]: These are already listed in the core dependencies and so don't need to be duplicated. Is the intent to remove them from the core deps list?

Comment (coryan) [reply ports/google-cloud-cpp/vcpkg.json]: Thanks for noticing that, and I am sorry it has taken me so many iterations to get this right.  Fixed, and rebased.  Tested each feature with x64-linux.

**Comment by JackBoosY** [comment]
Depends on https://github.com/microsoft/vcpkg/pull/19633.

**Comment by ras0219-msft** [comment]
Thanks!

**Pull request merged**